activerecord/lib/active_record/associations.rb

```diff
@@ -602,19 +602,24 @@ def association_instance_set(name, association)
       # you can also define callbacks that get triggered when you add an object to or remove an
       # object from an association collection.
       #
-      #   class Project
-      #     has_and_belongs_to_many :developers, after_add: :evaluate_velocity
+      #   class Firm < Company
+      #     has_many :clients, dependent: :destroy, after_add: :congratulate_client, after_remove: :log_after_remove
       #
-      #     def evaluate_velocity(developer)
+      #     def congratulate_client(record)
+      #       ...
+      #     end
+      #
+      #     def log_after_remove(record)
       #       ...
       #     end
-      #   end
       #
       # It's possible to stack callbacks by passing them as an array. Example:
       #
-      #   class Project
-      #     has_and_belongs_to_many :developers,
-      #                             after_add: [:evaluate_velocity, Proc.new { |p, d| p.shipping_date = Time.now}]
+      #   class Firm < Company
+      #     has_many :clients,
+      #              dependent: :destroy,
+      #              after_add: [:congratulate_client, Proc.new { |f, r| f.log << "after_adding#{r.id}" }],
```

Comment:
I think we should spell out the parameters here:

```suggestion
      #              after_add: [:congratulate_client, -> (firm, client) { "..." }],
```